### PR TITLE
Add recommendation to consider using dynamic credentials in security model

### PR DESCRIPTION
**Pull request by mpminardi**
### What
Adds section to `website/docs/cloud-docs/architectural-details/security-model.mdx`

### Why
Updates the TFC Security Model best practices to encourage the use of Dynamic Credentials over static credentials.

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [ ] Description links to related pull requests or issues, if any.

#### Content
- [ ] Redirects have been added to `website/redirects.js` for moved, renamed, or deleted pages.
- [ ] API documentation and the API Changelog have been updated. 
- [ ] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [ ] Pages with related content are updated and link to this content when appropriate.
- [ ] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [ ] New pages have metadata (page name and description) at the top.
- [ ] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [ ] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [ ] UI elements (button names, page names, etc.) are bolded.
- [ ] The Vercel website preview successfully deployed.

#### Reviews
- [ ] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/architectural-details/security-model.mdx [hunk 1]
```diff
@@ -154,3 +154,7 @@ Workspaces may be configured to share their state with other workspaces within t
...
+
+### Consider Using Dynamic Provider Credentials
+
+Storing static credentials in Terraform Cloud comes with increased risk in the event that these credentials are compromised. This is due to their long-lived nature, and the need to revoke these credentials in potentially many locations upon exposure. Using [Dynamic Provider Credentials](/terraform/cloud-docs/workspaces/dynamic-provider-credentials/) eliminates the need to store credentials in Terraform Cloud and mitigates the aforementioned risks by generating new temporary credentials for each run.
```
Comment (james-warren0) [inline website/docs/cloud-docs/architectural-details/security-model.mdx:160]: > Storing static credentials in Terraform Cloud comes with increased risk in the event that these credentials are compromised. This is due to their long-lived nature, and the need to revoke these credentials in potentially many locations upon exposure.

I'm being nitpicky on the language, but this feels more broad than we need to be. I think the specific scenario dynamic creds mitigates is a malicious user or compromised plan/apply (via injection or something else). Highlighting that dynamic credentials reduce the validity window of credentials so that in the event of a compromise of your plan or apply phase, those credentials are ephemeral and revoked at the end of the run, therefore reducing risk and exposure.

Comment (mpminardi) [reply website/docs/cloud-docs/architectural-details/security-model.mdx:160]: That's fair! Maybe the following then?

```suggestion
Storing static credentials in Terraform Cloud comes with increased risk in the event that these credentials are exposed by a malicious user or a compromised plan or apply operation. This is due to their long-lived nature, and the need to revoke these credentials in potentially many locations upon exposure. Using [Dynamic Provider Credentials](/terraform/cloud-docs/workspaces/dynamic-provider-credentials/) eliminates the need to store credentials in Terraform Cloud and reduces this potential risk and exposure by generating new temporary credentials for each run which expire after the run completes.
```

e.g., further scoping the first bit to add "by a malicious user or a compromised plan or apply operation."  and adjusting the end to be "and reduces this potential risk and exposure by generating new temporary credentials for each run which expire after the run completes."

Comment (james-warren0) [reply website/docs/cloud-docs/architectural-details/security-model.mdx:160]: I think that sounds great!

File: website/docs/cloud-docs/architectural-details/security-model.mdx [hunk 2]
```diff
@@ -154,3 +154,7 @@ Workspaces may be configured to share their state with other workspaces within t
 ### Treat Archivist URLs as Secrets
 
 Terraform Cloud uses a blob storage service called Archivist for storing various pieces of customer data. Archivist URLs (which have the origin <https://archivist.terraform.io>) are returned by various Terraform Cloud APIs, such as the [state versions API](/terraform/cloud-docs/api-docs/state-versions#fetch-the-current-state-version-for-a-workspace). Unlike the rest of Terraform Cloud’s API, Archivist’s API does not require the user to submit a Bearer Token with each request. Instead, Archivist URLs contain a signed short term authorization token that is used to perform authorization checks. For this reason, Archivist URLs must be treated as secrets and should not be logged or shared.
+
+### Consider Using Dynamic Provider Credentials
```
Comment (rkoron007) [inline website/docs/cloud-docs/architectural-details/security-model.mdx:158]: Could we shorten this to "Using dynamic credentials"? I think it reads a bit more clearly if folks aren't acquainted with "Dynamic Provider Credentials" yet, they still get from content what a dynamic credential is. 
```suggestion
### Using dynamic credentials
```

File: website/docs/cloud-docs/architectural-details/security-model.mdx [hunk 3]
```diff
@@ -154,3 +154,7 @@ Workspaces may be configured to share their state with other workspaces within t
...
+
+Storing static credentials in Terraform Cloud comes with increased risk in the event that these credentials are exposed by a malicious user or a compromised plan or apply operation. This is due to their long-lived nature, and the need to revoke these credentials in potentially many locations upon exposure. Using [Dynamic Provider Credentials](/terraform/cloud-docs/workspaces/dynamic-provider-credentials/) eliminates the need to store credentials in Terraform Cloud and reduces this potential risk and exposure by generating new temporary credentials for each run which expire after the run completes.
```
Comment (rkoron007) [inline website/docs/cloud-docs/architectural-details/security-model.mdx:160]: Adding suggestions for flow and clarity, and we don't capitalize feature names (just product names). Additionally, I assumed DPC works for both plans and applys so I changed the second sentence to "operation" instead of just "run". 
```suggestion
Storing static credentials in Terraform Cloud increases the inherent risk of a malicious user or a compromised plan or apply operation exposing your credentials. Because static credentials are usually long-lived and exposed in many locations, they are troublesome to revoke and replace.

Using [dynamic provider credentials](/terraform/cloud-docs/workspaces/dynamic-provider-credentials/) eliminates the need to store static credentials in Terraform Cloud, reducing the risk of exposure. Dynamic provider credentials generate new temporary credentials for each operation and expire after that operation completes. 
```

**Review by rkoron007 — APPROVED** [review]
Added some comments for style, but please feel free to push back on anything that doesn't feel good! ✨

File: website/docs/cloud-docs/architectural-details/security-model.mdx (hunk 2 above)
Comment (rkoron007) [reply website/docs/cloud-docs/architectural-details/security-model.mdx:158]: Dang, I just realized I deleted on my own comment by accident! Anyways, **thank you** for these very thoughtful questions! 

1. [Our style guide states to use sentence case for all headers](https://github.com/hashicorp/engineering-docs/blob/main/writing/style-guide.md#titles-and-headers), but that update _might_ have happened after this article was originally written. It's not a huge priority for older articles, but we should use sentence case from now on!  

(_My bad alert: When I first joined I didn't read this part of the style guide closely enough, so I made some suggestions to use title case in the past, which is totally my bad!_ 😅 ) 

If have the bandwidth and wouldn't mind, it'd be amazing if you updated the headers for the rest of the article too, but no problem if you'd rather not! 

2. You are totally correct! We should use the imperative in headers, so correcting my suggestion below:
```suggestion
### Use dynamic credentials
```

Comment (tylerwolf) [reply website/docs/cloud-docs/architectural-details/security-model.mdx]: Updated.  Want to take another look?

File: website/docs/cloud-docs/architectural-details/security-model.mdx [hunk 4]
```diff
@@ -1,46 +1,46 @@
 ---
-page_title: Security Model - Terraform Cloud
+page_title: Security model - Terraform Cloud
```
Comment (rkoron007) [inline website/docs/cloud-docs/architectural-details/security-model.mdx:2]: Sorry, I should have clarified that page metadata should stay title case:
```suggestion
page_title: Security Model - Terraform Cloud
```

**Review by rkoron007 — APPROVED** [review]
Thank you so much for sentence casing all the headers! One tiny suggestion, but everything else looks fantastic to me! ✅

File: website/docs/cloud-docs/architectural-details/security-model.mdx (hunk 4 above)
Comment (tylerwolf) [reply website/docs/cloud-docs/architectural-details/security-model.mdx]: Updated.  The style guide is a little confusing if that's that case.
> Use sentence case for page titles and all section headings on a page.
